tests/marray.test.ts

```diff
@@ -0,0 +1,35 @@
+import * as fc from 'fast-check';
+import { MArray } from '../src';
+
+describe('mArray', () => {
+  it('slice should work with randomly generated splice operations', () => {
+    const ARR_MAX_LENGTH = 1000;
+
+    fc.assert(
+      fc.property(
+        fc.array(fc.integer(), { maxLength: ARR_MAX_LENGTH }),
+        fc.array(
+          fc.record({
+            at: fc.integer(-ARR_MAX_LENGTH * 2, ARR_MAX_LENGTH * 2), // So the prob distrib a case where the index is out of bounds is between 0-50%
+            deleteCount: fc.integer(-ARR_MAX_LENGTH * 2, ARR_MAX_LENGTH * 2), // So the prob distrib a case where the count is out of bounds is between 0-50%
+            add: fc.array(fc.integer(), { maxLength: ARR_MAX_LENGTH }),
+          }),
+        ),
+        (data, splices) => {
+          const mArray = MArray.from(data);
+
+          for (const record of splices) {
+            const delsArray = data.splice(record.at, record.deleteCount, ...record.add);
+            const delsMArray = mArray.splice(record.at, record.deleteCount, ...record.add);
+
+            expect(delsMArray.toArray()).toEqual(delsArray);
+            expect(mArray.toArray()).toEqual(data);
+          }
+        },
+      ),
+      {
+        numRuns: 50,
```

Comment:
Does the default value (100) take too long?

Reply:
Turns out it doesn't -- only ~10 sec longer 😄 

Commit: [252c15a](https://github.com/hfour/yads/pull/33/commits/252c15a201fe14a9952deba2d1aaedf946c962f2)